[Bug]: Inline test errors displayed on wrong line

### Checklist

- [X] I have read and agree to the [Code of Conduct](https://github.com/CoderLine/mocha-vscode/blob/main/code-of-conduct.md)
- [X] I have searched for [related issues](https://github.com/CoderLine/mocha-vscode/issues) and [discussions](https://github.com/CoderLine/mocha-vscode/discussions) but they did not match this problem.
- [X] I want to contribute a bugfix myself.

### Actual behavior

<img width="1409" alt="image" src="https://github.com/user-attachments/assets/390b63e2-282f-4f38-9d4b-4f0fb97c2ffe" />

```js
AssertionError: expected [ { elementType: 'link', …(14) } ] to have a length of 5 but got 1
    at Context.<anonymous> file:///Users/albinekb/Devel/packages/browser-scripts/test/admin.test.ts:48:32    at runNextTicks (node:internal/process/task_queues:60:5)
    at process.processImmediate (node:internal/timers:449:9)
    at process.callbackTrampoline (node:internal/async_hooks:130:17)
```

The inline problem/output panel is displayed on the wrong line, it's always one line under the actual line.

### Expected behavior

The inline error should be on the expected line, not the line under it.

As the AssertionError correctly outputs line 48 in this case.

### Minimal, Reproducible Example

I can setup a repro/test when I got time to look into fixing this bug

### Output

2024-12-22 01:37:54.547 [info] Reloaded tests from '/Users/albinekb/Devel/apps/browser-engine/node_modules/browser-scripts/test/admin.test.ts' Suite: +0 ~1 -0; Test: +0 ~1 -0
2024-12-22 01:37:54.666 [info] Reloaded tests from '/Users/albinekb/Devel/packages/browser-scripts/test/admin.test.ts' Suite: +0 ~1 -0; Test: +0 ~1 -0
2024-12-22 01:37:55.683 [info] Start test execution with args ["/Users/albinekb/.local/state/fnm_multishells/85518_1734821474670/bin/node","/Users/albinekb/Devel/node_modules/.pnpm/mocha@10.8.2/node_modules/mocha/bin/mocha.js","--config","/Users/albinekb/Devel/packages/browser-scripts/.mocharc.json","--reporter","/Users/albinekb/Devel/node_modules/.pnpm/mocha@10.8.2/node_modules/.mocha-vscode/fullJsonStreamReporter.js","--ignore","**/*.*","--file","test/admin.test.ts","--grep","/^(admin should pass$)/"]
2024-12-22 01:37:55.683 [info] Start test execution with env {}
2024-12-22 01:37:58.998 [error] Error: Test process exited with code 1
    at ChildProcess.<anonymous> (/Users/albinekb/.cursor/extensions/coderline.mocha-vscode-1.2.4/out/extension.js:592:708)
    at ChildProcess.emit (node:events:519:28)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:294:12)

### Plugin Version Details

Name: Mocha for VS Code
Id: coderline.mocha-vscode
Description: Run and debug Mocha tests right within VS Code.
Version: 1.2.4
Publisher: coderline
VS Marketplace Link: https://marketplace.cursorapi.com/items?itemName=coderline.mocha-vscode

### VS Code Version Details

<details>
<summary>Details</summary>



Tested in cursor and vscode:

Version: 1.96.0-insider
Commit: 3551cb01fa6f3a838e2d160df704cc3debfb9896
Date: 2024-10-30T05:04:18.074Z
Electron: 32.2.1
ElectronBuildId: 10427718
Chromium: 128.0.6613.186
Node.js: 20.18.0
V8: 12.8.374.38-electron.0
OS: Darwin arm64 24.0.0

Cursor
Version: 0.44.5
VSCode Version: 1.93.1
Commit: 1d610252e6812bf33245763f0742a534fd0f1d90
Date: 2024-12-20T00:02:28.554Z
Electron: 30.5.1
Chromium: 124.0.6367.243
Node.js: 20.16.0
V8: 12.4.254.20-electron.0
OS: Darwin arm64 24.0.0
</details>

### Further details
https://github.com/CoderLine/mocha-vscode/blob/2e20980c86028e63b5fae1c2eefae95d01fb0787/src/source-map.ts#L32-L45

--- a/src/runner.ts
+++ b/src/runner.ts
@@ -707,9 +707,12 @@ async function deriveSourceLocation(
 
   const maintainer = store.maintain(fileUri);
   const mapping = await (maintainer.value || maintainer.refresh());
+  // in parsed stack traces line and column numbers are 1-based, in the VS Code APIs lines are 0-based
+  const zeroBasedLine = Number(line) - 1;
+  const zeroBasedCol = Number(col) - 1;
   const value =
-    mapping?.originalPositionFor(Number(line), Number(col)) ||
-    new vscode.Location(fileUri, new vscode.Position(Number(line), Number(col)));
+    mapping?.originalPositionFor(zeroBasedLine, zeroBasedCol) ||
+    new vscode.Location(fileUri, new vscode.Position(zeroBasedLine, zeroBasedCol));
 
   // timeout the maintainer async so that it stays alive for any other immediate teset usage in the file:
   setTimeout(() => maintainer.dispose(), 5000);

--- a/src/source-map.ts
+++ b/src/source-map.ts
@@ -30,6 +30,7 @@ export const identityMapping = (file: vscode.Uri): IMappingAccessor => ({
 });
 
 const smMappingAccessor = (file: vscode.Uri, sm: TraceMap): IMappingAccessor => ({
+  // @jridgewell/trace-mapping: Lines start at line 1, columns at column 0.
   originalPositionFor(line, column) {
     const {
       source,

--- a/src/test/integration/simple.test.ts
+++ b/src/test/integration/simple.test.ts
@@ -235,8 +235,8 @@ describe('simple', () => {
     expect(failed.message?.location).to.not.be.undefined;
     expect(failed.message?.location?.uri.toString()).to.include('hello.test.js');
     expect(path.isAbsolute(failed.message!.location!.uri.fsPath)).to.be.true;
-    expect(failed.message?.location?.range.start.line).to.equal(12);
-    expect(failed.message?.location?.range.start.character).to.equal(5);
+    expect(failed.message?.location?.range.start.line).to.equal(11);
+    expect(failed.message?.location?.range.start.character).to.equal(4);
   });
 
   it('handles file and directory excludes', async () => {
@@ -318,10 +318,10 @@ describe('simple', () => {
     });
 
     const failed = run.states.find((s) => s.state === 'failed');
-    expect(failed!.message!.location!.range.start.line).to.equal(4);
-    expect(failed!.message!.location!.range.start.character).to.equal(11);
-    expect(failed!.message!.location!.range.end.line).to.equal(4);
-    expect(failed!.message!.location!.range.end.character).to.equal(11);
+    expect(failed!.message!.location!.range.start.line).to.equal(3);
+    expect(failed!.message!.location!.range.start.character).to.equal(10);
+    expect(failed!.message!.location!.range.end.line).to.equal(3);
+    expect(failed!.message!.location!.range.end.character).to.equal(10);
     expect(failed!.message!.location!.uri.fsPath).to.equal(item.uri!.fsPath);
   });
 });

--- a/src/test/integration/typescript.test.ts
+++ b/src/test/integration/typescript.test.ts
@@ -84,7 +84,7 @@ describe('typescript', () => {
     expect(failed.message?.location).to.not.be.undefined;
     expect(failed.message?.location?.uri.toString()).to.include('hello.test.ts');
     expect(path.isAbsolute(failed.message!.location!.uri.fsPath)).to.be.true;
-    expect(failed.message?.location?.range.start.line).to.equal(29);
-    expect(failed.message?.location?.range.start.character).to.equal(5);
+    expect(failed.message?.location?.range.start.line).to.equal(28);
+    expect(failed.message?.location?.range.start.character).to.equal(4);
   });
 });

